### PR TITLE
feat: 支持自定义节点多面板

### PR DESCRIPTION
**Pull request by draco-china**
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Enhancement (changes that improvement of current feature or performance)
- [ ] Refactoring (changes that neither fixes a bug nor adds a feature)
- [ ] Test Case (changes that add missing tests or correct existing tests)
- [ ] Code style optimization (changes that do not affect the meaning of the code)
- [x] Docs (changes that only update documentation)
- [ ] Chore (changes that don't modify src or test files)

### Self Check before Merge

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/antvis/x6/blob/master/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by draco-china** [comment]
> @draco-china Please provide us with more info about this pull request.

将registerNode 类型改为 registerNode[] ,原本只支持一个自定义面板，将之改为多个，并且已修改相关逻辑

File: packages/xflow-extension/src/flowchart-node-panel/interface.tsx [hunk 1]
```diff
@@ -44,16 +44,20 @@ export interface IFlowchartNode {
...
   show?: boolean
   showHeader?: boolean
+  showOfficial?: boolean
   /** 自定义节点 */
   registerNode?: {
     title?: string
+    key: string
     nodes: ICustomNode[]
-  }
+  }[]
```
Comment (lyn-boyu) [inline packages/xflow-extension/src/flowchart-node-panel/interface.tsx:60]: this is a break change
这样写之前的组件会直接报错吧？是不是可以兼容下非数组的情况？
内部都使用数组渲染，如果非数组就把他包成一个长度为1的数组

File: packages/xflow-extension/src/flowchart-node-panel/utils.tsx [hunk 2]
```diff
@@ -77,7 +80,23 @@ export const nodeService = async nodes => {
   ]
 }
 
-export const registerCustomNode = (nodes = []) => {
+export const registerCustomNode = (
+  props: {
```
Comment (lyn-boyu) [inline packages/xflow-extension/src/flowchart-node-panel/utils.tsx:84]: props => panelConfigs

File: packages/xflow-extension/src/flowchart-node-panel/interface.tsx (hunk 1 above)
Comment (draco-china) [reply packages/xflow-extension/src/flowchart-node-panel/interface.tsx:60]: 的确这样会更好

**Review by lxfu1 — APPROVED** [review]

**Pull request merged**